airflow/api_fastapi/core_api/datamodels/task_instances.py

```diff
@@ -83,3 +91,26 @@ class TaskDependencyCollectionResponse(BaseModel):
     """Task scheduling dependencies collection serializer for responses."""
 
     dependencies: list[TaskDependencyResponse]
+
+
+class TaskInstancesBatchBody(BaseModel):
+    """Task Instance body for get batch."""
+
+    dag_ids: list[str] | None = None
+    dag_run_ids: list[str] | None = None
+    task_ids: list[str] | None = None
+    state: list[TaskInstanceState | None] | None = None
+    logical_date_gte: AwareDatetime | None = None
+    logical_date_lte: AwareDatetime | None = None
```

Comment:
Makes me realize that all `datetime` in the API should be `AwareDatetime` I believe. (We don't know what to do with naive datetime, and assuming UTC is prone to errros)

Reply:
TI will always be in UTC since scheduler operators on UTC. You could use `airflow.api_fastapi.common.types.UtcDateTime`

Reply:
But you are right, tough to expect users to send date in UTC, so maybe accept AwareDatatime and convert to UTC?

Reply:
I think that as long as the timezone is specified, there is no ambiguity. Because comparison is correct in the server, and then when persisted to the db it gets converted to UTC (at least that's what I remember from postgres).

Reply:
But I needed that for a specific test to succeed. (Timezone naive datetime should be refused, that's what legacy behavior is implementing).

I'll follow up with a specific PR on that matter.